src/relay/libp2p_transport_relay.erl

```diff
@@ -143,7 +144,10 @@ connect_rcv(Swarm, MAddr, SAddress, SessionPid) ->
         {error, _Reason}=Error ->
             true = libp2p_config:remove_relay_sessions(libp2p_swarm:tid(Swarm), SAddress),
             lager:error("no relay sessions ~p", [_Reason]),
-            Error
+            Error;
+        {'DOWN', _Ref, process, _Stream, _Reason} ->
```

Comment:
You should pass in the stream pid or the monitor ref here so you can match on them rather than just a random DOWN message.